catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java

```diff
@@ -117,18 +118,37 @@ public List<String> listTables(String databaseName) throws NoSuchSchemaException
 
   @Override
   public JdbcTable load(String databaseName, String tableName) throws NoSuchTableException {
+    // We should handle case sensitivity and wild card issue in some catalog tables, take a
+    // MySQL table for example.
+    // 1. MySQL will get table 'a_b' and 'A_B' when we query 'a_b' in a case-insensitive charset
+    // like utf8mb4.
+    // 2. MySQL will view 'a_b' as a wild card, and it will match any table name that starts with
+    // 'a',then any character and then 'b'.
     try (Connection connection = getConnection(databaseName)) {
       // 1.Get table information
-      ResultSet table = getTable(connection, databaseName, tableName);
-      if (!table.next() || !tableName.equals(table.getString("TABLE_NAME"))) {
+      ResultSet table = getTable(connection, escapeSQL(databaseName), escapeSQL(tableName));
+      // The result of tables may be more than one due to the reason above, so we need to check the
+      // result
+      JdbcTable.Builder jdbcTableBuilder = new JdbcTable.Builder();
+      boolean found = false;
+      while (table.next()) {
+        if (Objects.equals(table.getString("TABLE_NAME"), tableName)) {
```

Comment:
is it possible, I create with a DB with name `A`, the underlying RDMS trans the name to `a`?  should we consider this?

Reply:
It's possible, but I'm not going to put the logic in the base class. Any JDBC catalog that is case-insensitive in the table name can overwrite this method, AFAIK, most JDBC tables are case-sensitive here.